binding.c

```diff
@@ -458,10 +734,28 @@ NAPI_METHOD(udx_napi_init) {
   return NULL;
 }
 
-NAPI_METHOD(udx_napi_socket_init) {
-  NAPI_ARGV(7)
-  NAPI_ARGV_BUFFER_CAST(udx_napi_t *, udx, 0)
-  NAPI_ARGV_BUFFER_CAST(udx_napi_socket_t *, self, 1)
+napi_value
+udx_napi_socket_init (napi_env env, napi_callback_info info) {
+  napi_value argv[7];
+  size_t argc = 7;
+  if (napi_get_cb_info(env, info, &argc, argv, NULL, NULL) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_cb_info(env, info, &argc, argv, NULL, NULL) failed!");
+    return NULL;
+  }
+
+  udx_napi_t *udx;
+  size_t udx_len;
+  if (napi_get_buffer_info(env, argv[0], (void **) &udx, &udx_len) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_buffer_info(env, argv[0], (void **) &udx, &udx_len) failed!");
+    return NULL;
+  }
+
+  udx_napi_socket_t *self;
+  size_t self_len;
+  if (napi_get_buffer_info(env, argv[1], (void **) &self, &self_len) != napi_ok) {
+    napi_throw_error(env, NULL, "napi_get_buffer_info(env, argv[1], (void **) &self, &self_len) failed!");
```

Comment:
just ignore all the throws here and possible errors from napi_get_buffer_info in these 3 lines - will never happen in practice